PSEUDO API 1.md

```diff
@@ -0,0 +1,166 @@
+
+## XR
+
+	interface XR {
+		Promise<sequence<XRDisplay>> getDisplays();
+	};
+
+
+## XRDisplay
+
+	interface XRDisplay {
+		readonly attribute DOMString displayName;
+		readonly attribute boolean isExternal;
+
+		Promise<boolean> supportsSession(XRSessionCreateParametersInit parameters);
+		Promise<XRSession> requestSession(XRSessionCreateParametersInit parameters);
+
+		attribute EventHandler ondeactivate;
+	};
+
+### Todo
+
+- calibration
+
+## Reality
+
+	interface Reality {
+		readonly atttribute DOMString realityName;
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRCoordinates stageLocation;
+
+		Promise<boolean> requestStageLocation(float x, float y float z, XRCoordinates)
+		Promise<boolean> requestResetStageLocation()
+
+		Promise<XRPointCloud> getPointCloud()
+
+		Promise<XRLightEstimate> getLightEstimate()
+
+		attribute EventHandler onchange;
+	};
+
+### Todo
+
+- configuration
+- anchors
+- manifold instead of point cloud
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute float ambientIntensity;
+		readonly attribute float ambientColorTemperature;
+	}
+
+## XRSession
+
+	interface XRSession : EventTarget {
+		readonly attribute XRDisplay display;
+		readonly attribute XRSessionCreateParameters createParameters;
+
+		attribute double depthNear;
+		attribute double depthFar;
+
+		attribute XRLayer layer;
+
+		Promise<sequence <Reality>> getRealities();
+
+		Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceType type);
+
+		long requestFrame(XRFrameRequestCallback callback);
+		void cancelFrame(long handle);
+
+		Promise<void> endSession();
+
+		attribute EventHandler onblur;
+		attribute EventHandler onfocus;
+		attribute EventHandler onresetpose;
+		attribute EventHandler onended;
+	};
+
+## XRPresentationFrame
+
+	interface XRPresentationFrame {
+		readonly attribute FrozenArray<XRView> views;
+
+		XRDisplayPose? getDisplayPose(XRCoordinateSystem coordinateSystem);
```

Comment:
In argon.js, we expose them by name as Cesium Entities.  There are a fixed set of predefined ones (ar.user, ar.stage, ar.origin, ar.floor, etc) and then new ones are created depending on the platform and tracking tech (e.g., vuforia.stonesandchip.stones, etc)

I don't know how this maps with Windows Holographic, or to ARKit.  Tango has no in-built capabilities, it's all app-level post processing, so we can do what we want there.